csharp/src/Apache.Arrow.Flight/FlightHandshakeRequest.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf;
+
+namespace Apache.Arrow.Flight;
+
+public class FlightHandshakeRequest
```

Comment:
The Flight.Proto is imported as internal and not visible outside of the library.  In order for an application to implement the `Handshake` method it would need access to a public version of `FlightHandshakeResult` or a client to create a `HandshakeRequest` it will need access to a public class that translates into an internal proto defined class.  

This is how the .Net library has implements other features for `Flight`.  For example `FlightCriteria` for `ListFlights` has a similar wrapper.  Wrappers like this can be found in use for most of the `FlightFerver` abstract functions.  I'm not a fan of the paradigm myself, however, I felt flipping the internal protos to public was a larger item than I was willing to tackle for this PR.

Reply:
Ok, this makes sense.  There are some bad things that happen if we expose the internal protobuf objects in other languages (e.g. C++) so it's probably motivated from there.  I don't know if these same problems happen in C# or not.  However, I agree it should not be the concern of this PR.